webrender/src/renderer.rs

```diff
@@ -35,19 +35,65 @@ use std::sync::{Arc, Mutex};
 use std::sync::mpsc::{channel, Receiver, Sender};
 use std::thread;
 use texture_cache::{BorderType, TextureCache, TextureInsertOp};
-use tiling::{self, Frame, FrameBuilderConfig, GLYPHS_PER_TEXT_RUN, PrimitiveBatchData, PackedTile};
-use tiling::{TransformedRectKind, RenderTarget, ClearTile, PackedStackingContext};
+use tiling::{self, Frame, FrameBuilderConfig, GLYPHS_PER_TEXT_RUN, PrimitiveBatchData};
+use tiling::{TransformedRectKind, RenderTarget, ClearTile};
 use time::precise_time_ns;
 use webrender_traits::{ColorF, Epoch, PipelineId, RenderNotifier};
 use webrender_traits::{ImageFormat, MixBlendMode, RenderApiSender};
 use offscreen_gl_context::{NativeGLContext, NativeGLContextMethods};
 
 pub const BLUR_INFLATION_FACTOR: u32 = 3;
 pub const MAX_RASTER_OP_SIZE: u32 = 2048;
+pub const MAX_VERTEX_TEXTURE_WIDTH: usize = 1024;
 
-const UBO_BIND_TILES: u32 = 0;
 const UBO_BIND_DATA: u32 = 1;
 
+struct VertexDataTexture {
+    id: TextureId,
+}
+
+impl VertexDataTexture {
+    fn new(device: &mut Device) -> VertexDataTexture {
+        let id = device.create_texture_ids(1)[0];
+
+        VertexDataTexture {
+            id: id,
+        }
+    }
+
+    fn init<T: Default>(&mut self,
+                        device: &mut Device,
+                        data: &mut Vec<T>) {
+        let item_size = mem::size_of::<T>();
+        debug_assert!(item_size % 16 == 0);
+        let vecs_per_item = item_size / 16;
```

Comment:
In a RGBAF32 texture, 16 bytes is one texture fetch / texel. All our current data matches this requirement (which was the 16 byte UBO alignment requirement).